crates/wasi-common/src/snapshots/preview_1.rs

```diff
@@ -560,12 +571,17 @@ impl wasi_snapshot_preview1::WasiSnapshotPreview1 for WasiCtx {
     }
 
     async fn fd_sync(&mut self, fd: types::Fd) -> Result<(), Error> {
-        self.table()
+        // Check if fd is a file, and has rights, just to reject those cases
+        // with the errors expected:
+        let _ = self
+            .table()
             .get_file(u32::from(fd))?
-            .get_cap(FileCaps::SYNC)?
-            .sync()
-            .await?;
-        Ok(())
+            .get_cap(FileCaps::SYNC)?;
+        // This operation from cloud abi is linux and mac-specific, though it has different
+        // semantics on those two platforms. WASI never specified the semantics, and our test suite
+        // never included this operation. We have removed this operation from preview 2, and we are
+        // no longer supporting it in preview 1 as well.
+        Err(Error::not_supported())
```

Comment:
This is still removing support for `fd_sync` and `fd_datasync`, which I think we should keep. Even though they're not portable, there is value in being able to expose what functionality the host might have to applications.

Reply:
I got rid of that change locally but I didnt hit `-f` when I pushed it! Sorry.